c_regex/regex.c

```diff
@@ -39,23 +39,28 @@ struct TestCase {
     int isMatch;
 };
 
-void free_lines(char **lines, int size) {
-    for (int i = 0; i < size; i++)
-        free(lines[i]);
-    free(lines);
+void free_lines(char **lines, int n) {
```

Comment:
`n` means the every `lines[i]` (0 <= i < n) are initialized aready. 

The reason why I don't use `memset` or `for-loop` to initialize array is that it will make this demo a bit longer.